### PR TITLE
Arsenal - Fix displaying virtual items & weapon attachments in cargo

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Unique items would not show up in containers when they were available in the arsenal. E.g. If you put NVGs in your backpack, they wouldn't show up in the backpack's items when using an arsenal that has those NVGs available in the NVG tab.
- If a unit had weapons in a container, it would show all of its accessories as unique items. Now you no longer see a weapon's accessories.

This essentially reverts some of the changes I made in #9040, but it's for the better.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/functions/fnc_fillRightPanel.sqf [hunk 1]
```diff
@@ -409,16 +409,16 @@ if (_isContainer) then {
...
+            _containerItems = backpackItems GVAR(center);
 
             backpackContainer GVAR(center)
         };
     };
 
     // Find out how many items of a type there are and update the number displayed
     for "_lbIndex" from 0 to (lnbSize _ctrlPanel select 0) - 1 do {
-        _ctrlPanel lnbSetText [[_lbIndex, 2], str (_containerItems getOrDefault [_ctrlPanel lnbData [_lbIndex, 0], 0])];
+        _ctrlPanel lnbSetText [[_lbIndex, 2], str ({_ctrlPanel lnbData [_lbIndex, 0] == _x} count _containerItems)];
```
Comment (rautamiekka) [inline addons/arsenal/functions/fnc_fillRightPanel.sqf:421]:
```suggestion
        private _xItem = _ctrlPanel lnbData [_lbIndex, 0];
        _ctrlPanel lnbSetText [[_lbIndex, 2], str ({_xItem == _x} count _containerItems)];
```

File: addons/arsenal/functions/fnc_updateUniqueItemsList.sqf [hunk 2]
```diff
@@ -255,7 +261,13 @@ private _itemInfoType = 0;
                 // Unknown
                 default {
                     // Don't add items that are part of the arsenal
-                    if !(_x in GVAR(virtualItemsFlatAll)) then {
+                    if (
+                        !(_x in ((GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS) get IDX_VIRT_OPTICS_ATTACHMENTS)) &&
+                        {!(_x in ((GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS) get IDX_VIRT_FLASHLIGHT_ATTACHMENTS))} &&
+                        {!(_x in ((GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS) get IDX_VIRT_MUZZLE_ATTACHMENTS))} &&
+                        {!(_x in ((GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS) get IDX_VIRT_BIPOD_ATTACHMENTS))} &&
+                        {!(_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS))}
```
Comment (rautamiekka) [inline addons/arsenal/functions/fnc_updateUniqueItemsList.sqf:269]:
```suggestion
                    private _attachments = GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS;
                    if (
                        !(_x in (_attachments get IDX_VIRT_OPTICS_ATTACHMENTS)) &&
                        {!(_x in (_attachments get IDX_VIRT_FLASHLIGHT_ATTACHMENTS))} &&
                        {!(_x in (_attachments get IDX_VIRT_MUZZLE_ATTACHMENTS))} &&
                        {!(_x in (_attachments get IDX_VIRT_BIPOD_ATTACHMENTS))} &&
                        {!(_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS))}
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]
Performance improvements.

**Comment by LinkIsGrim** [comment]
> Unique items would not show up in containers when they were available in the arsenal. E.g. If you put NVGs in your backpack, they wouldn't show up in the backpack's items when using an arsenal that has those NVGs available in the NVG tab.

Will those items be treated as unique or virtual items within the container (can their quantity be increased or only decreased?)

> If a unit had weapons in a container, it would show all of its accessories as unique items. Now you no longer see a weapon's accessories.

Good change.

**Comment by johnb432** [comment]
> Will those items be treated as unique or virtual items within the container (can their quantity be increased or only decreased?)

As unique items, so only decreased.

**Comment by LinkIsGrim** [comment]
> > Will those items be treated as unique or virtual items within the container (can their quantity be increased or only decreased?)
> 
> As unique items, so only decreased.

Not a fan, but I understand this was pre-#9040 behavior. I'm okay with this as is, but being able to increase quantity of those items would be good UX, IMO (though it's added complexity).

**Review by LinkIsGrim — APPROVED** [review]

**Comment by LinkIsGrim** [comment]
> I'm okay with this as is, but being able to increase quantity of those items would be good UX, IMO (though it's added complexity).

I have an idea on how to do this without increasing complexity too much. Separate PR though.

**Pull request merged**